provisioner/chef-client/provisioner.go

```diff
@@ -194,6 +196,10 @@ func (p *Provisioner) Prepare(raws ...interface{}) error {
 			errs, fmt.Errorf("server_url must be set"))
 	}
 
+	if p.config.ChefLicense == "" {
+		p.config.ChefLicense = "accept-silent"
```

Comment:
I think it may be a good idea to leave this field blank unless specified.
It might be a breaking change in the sense that old version of Chef could error when we set `ChefLicense` ? Also I think in terms of license it would be better to have to set it explicitly in a buildfile.

Reply:
( other than that LGTM !! :) )

Reply:
@azr The chef license was added to older chef version as a no-op. But kinda agree I was just going for a mode that just works with existing users. The transition to chef 15 happened  automatically as part of the omnibus install process and broke pretty much every existing chef packer provision without them modifying the install command. Hence setting a default for this. 

I'm would like to add a feature to the chef provisioner to be able to specify the chef version which would help with these types of changes in future since you would be able to temporarily lock the chef version until a fix is pushed out

Reply:
Hey @aaronwalker, ah thanks for the explanation, I'm a chef newbie !

Reply:
Hey @aaronwalker, ah thanks for the explanation, I'm a chef newbie but I think it would be nice to have it work for everyone.
An easy thing we could do is to programatically parse the chef version and default the `ChefLicense` value based on it ? Is that possible ?

Reply:
@azr I might be possible via the installer to query the version but unless someone has overridden the install_command and the have set `skip_install` to true it will install the latest stable version which is now chef 15. Also we are adding `chef_license` to the client.rb template which is ignored by older versions anyway. So having in there as no with a default value has no affect.

Maybe we should only set a default if skip_install is false and install_command hasn't been overridden?

Reply:
> Maybe we should only set a default if skip_install is false and install_command hasn't been overridden?

I agree ! 🙂